### PR TITLE
Brians branch

### PR DESCRIPTION
**Pull request by Brian-Doucet**
Added a bunch of new features.

1.  Added some functions for parsing specific HTML from the property details page for a specific listing
2. Added a new BaseModel to models.py for Address
3. Updated BaseModel for ZillowData with additional fields which come from the functions in utils
4. Added a sample HTML file from a listing for use when testing. This way we're not sending repeated requests

TODO:
Need to incorporate the new functions from `utils.py` into the ZillowScraper() class. I wasn't sure how to add in the Address model.

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    Args:
+        url (str): Zillow URL for a specific listing
+
+    Returns:
+        str: Zillow property id
+    """
+    pattern = re.compile(r"(\d+)_zpid")
+    zpid = re.search(pattern, url).group(1)
+
+    return zpid
```
Comment (ilyayudkovich) [inline utils.py:41]: i like this

File: models/zillow.py [hunk 2]
```diff
@@ -6,27 +6,35 @@
...
         return self.__fields__
 
 
-class Address(BaseModel):
-    """
-    TODO: capture all the fields based on the following data
-    'address': {'@type': 'PostalAddress', '@context': 'http://schema.org', 'streetAddress': '3903 Perkins Rd', 'addressLocality': 'Thompsons Station', 'addressRegion': 'TN', 'postalCode': '37179'}
-    """
-    pass
+class ZillowAddress(BaseModel):
```
Comment (ilyayudkovich) [inline models/zillow.py]: #question: is this going to be part of ZillowData?

File: zillow.py [hunk 3]
```diff
@@ -51,16 +53,37 @@ def get_property_details(self, url: str) -> ZillowData:
...
+            property_url=home_details.get("url"),
+            year_built=facts_features.get("Year built"),
+            heating=facts_features.get("Heating"),
+            cooling=facts_features.get("Cooling"),
+            parking=facts_features.get("Parking"),
+            lot_size=facts_features.get("Lot"),
+            price_per_sqft=facts_features.get("Price/sqft"),
+            hoa_dues=facts_features.get("HOA"),
         )
+        # Not sure how to incorporate this object
```
Comment (ilyayudkovich) [inline zillow.py]: you can make this part of ZillowData object to keep it all in one place

File: models/zillow.py (hunk 2 above)
Comment (Brian-Doucet) [reply models/zillow.py]: Yeah I really wanted to include all of the address data in the csv file.

**Pull request merged**